constellations/gemini-admin/src/ask_for_help/afhEdit.js

```diff
@@ -0,0 +1,234 @@
+            window.location.hash = "afh";
+        });
+    }
+
+    onSavedOk() {
+        this.onDismissModal();
+        window.location.hash = "afh";
+    }
+
+    onDismissModal() {
+        this.setState({
+            showDeleteModal: false,
+            showSaveModal: false,
+        });
+    }
+
+    onMomentChange(newMoment) {
+        this.setState({ inputPostedAt: newMoment, date: newMoment });
+    }
+
+    render() {
+        const title =
+            (this.state.isEdit ? "Edit" : "Add") + " Ask For Help Session";
+
+        const optLocations = this.state.locations.map((loc, index) => (
+            <option key={index}>{loc.locationId}</option>
+        ));
+
+        let deleteButton = <div></div>;
+        if (this.state.isEdit) {
+            deleteButton = (
+                <button className="btn-delete" onClick={this.onClickDelete}>
+                    Delete
+                </button>
+            );
+        }
+
+        let modalDiv;
+        let modalContent;
+        let showModal;
+        if (this.state.showDeleteModal) {
+            showModal = this.state.showDeleteModal;
+            modalContent = (
+                <YesNoModal
+                    text={"Are you sure you want to delete?"}
+                    onAccept={this.onConfirmDelete}
+                    onReject={this.onDismissModal}
+                />
+            );
+        }
+        if (this.state.showSaveModal) {
+            showModal = this.state.showSaveModal;
+            modalContent = (
+                <OkayModal
+                    text={"Ask for help information saved!"}
+                    onOkay={this.onSavedOk}
+                />
+            );
+        }
+        if (modalContent) {
+            modalDiv = (
+                <Modal
+                    content={modalContent}
+                    show={showModal}
+                    onDismiss={this.onDismissModal}
+                />
+            );
+        }
+
+        return (
+            <div id="view-afh-edit">
+                {modalDiv}
+                <h2>{title}</h2>
+
+                <AFHEditDateTime
+                    postedAt={this.state.date}
+                    onMomentChange={this.onMomentChange}
+                />
+
+                <h3>Time</h3>
+                <input
+                    value={this.state.timeString}
+                    onChange={(e) => this.handleChange(e, "timeString")}
+                />
+
+                <h3>Title</h3>
+                <input
+                    value={this.state.title}
+                    onChange={(e) => this.handleChange(e, "title")}
+                />
+
+                <h3>Subject</h3>
+                <select
+                    value={this.state.subject}
+                    onChange={(e) => this.handleChange(e, "subject")}>
+                    <option>Math</option>
+                    <option>English</option>
+                    <option>Computer Programming</option>
```

Comment:
usually you want to make the option label and value different. So something like this:

```
<option value="math">Math</option>
<option value="computer">Computer Programming</option>
```

We want the value to be all lower-case so we keep values consistent and not have to worry about case sensitivity and it gives us the option to be flexible with our user-facing labels. So in other words, we want our data to be short and "matter-of-fact" and we can change the displayed label without changing the underlying data!

Reply:
Orion returns "error 01: invalid subject" when I change it to lowercase